File: backend/mijung/src/main/java/com/example/mijung/material/entity/Material.java
package com.example.mijung.material.entity;

import com.example.mijung.ingredient.entity.Ingredient;
import com.example.mijung.recipe.entity.Recipe;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Material {

  @Id
  @Column(name="material_id", updatable=false, nullable=false)
  private Integer id;

  @Column(updatable=false, nullable=false, columnDefinition = "TEXT")
  private String name;

  @Column(updatable = false, columnDefinition = "TEXT")
  private String capacity;

  @Column(updatable = false, nullable = false, columnDefinition = "TEXT")
  private String type;

  @Column(updatable = false, nullable = false)
  private Boolean analyzed;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name="recipe_id", updatable = false, nullable = false)
  private Recipe recipe;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name="ingredient_id", updatable = false)
  private Ingredient ingredient;
}

File: fastapi/app/schemas/material.py
from typing import Optional
from pydantic import BaseModel, ConfigDict
from sqlalchemy import TEXT, Column, ForeignKey, Integer, String, Boolean
from sqlalchemy.orm import relationship
from .base import Base

class Material(Base):
    __tablename__ = 'Material'
    
    material_id: int = Column(Integer, primary_key=True, nullable=False)
    name: str = Column(TEXT, nullable=False)
    capacity: Optional[str] = Column(TEXT, nullable=True)
    type: str = Column(TEXT, nullable=False)
    analyzed: bool = Column(Boolean, nullable=False)

    recipe_id: int = Column(Integer, ForeignKey('Recipe.recipe_id'), nullable=False)
    ingredient_id: Optional[int] = Column(Integer, ForeignKey('Ingredient.ingredient_id'), nullable=True)

    recipe = relationship("Recipe", back_populates="material")  # Adjust back_populates accordingly
    ingredient = relationship("Ingredient", back_populates="material")  # Adjust back_populates accordingly

class MaterialCreate(BaseModel):
    name: str
    capacity: Optional[str] = None
    type: str
    analyzed: bool
    recipe_id: int
    ingredient_id: Optional[int] = None

class MaterialRead(BaseModel):
    id: int
    name: str
    capacity: Optional[str] = None
    type: str
    analyzed: bool
    recipe_id: int
    ingredient_id: Optional[int] = None

    model_config = ConfigDict(from_attributes=True)